appservice/src/github/connectToGitHub.ts

```diff
@@ -38,35 +44,39 @@ export async function connectToGitHub(node: AzureTreeItem, client: SiteClient, c
         throw new Error(noToken);
     }
 
-    await signRequest(requestOptions, new TokenCredentials(oAuth2Token));
-    requestOptions.url = 'https://api.github.com/user';
-    const gitHubUser: gitHubOrgData = await getJsonRequest<gitHubOrgData>(requestOptions, node, client, context);
-    requestOptions.url = 'https://api.github.com/user/orgs';
-    const gitHubOrgs: gitHubOrgData[] = await getJsonRequest<gitHubOrgData[]>(requestOptions, node, client, context);
-    const orgQuickPicks: IAzureQuickPickItem<gitHubOrgData>[] = createQuickPickFromJsons([gitHubUser], 'login').concat(createQuickPickFromJsons(gitHubOrgs, 'login'));
-    const orgQuickPick: gitHubOrgData = (await ext.ui.showQuickPick(orgQuickPicks, { placeHolder: 'Choose your organization.' })).data;
-    requestOptions.url = nonNullProp(orgQuickPick, 'repos_url');
-
-    const picksCache: ICachedQuickPicks<gitHubReposData> = { picks: [] };
-    let repoQuickPick: gitHubReposData | undefined;
-    do {
-        repoQuickPick = (await ext.ui.showQuickPick(getGitHubReposQuickPicks(picksCache, requestOptions, node, client, context), { placeHolder: 'Choose repository.' })).data;
-    } while (!repoQuickPick);
+    await signRequest(context.requestOption, new TokenCredentials(oAuth2Token));
 
-    requestOptions.url = `${repoQuickPick.url}/branches`;
-    const gitHubBranches: gitHubBranchData[] = await getJsonRequest<gitHubBranchData[]>(requestOptions, node, client, context);
-    const branchQuickPicks: IAzureQuickPickItem<gitHubBranchData>[] = createQuickPickFromJsons(gitHubBranches, 'name');
-    const branchQuickPick: gitHubBranchData = (await ext.ui.showQuickPick(branchQuickPicks, { placeHolder: 'Choose branch.' })).data;
+    const title: string = localize('connectGitHubRepo', 'Connect GitHub repository');
+    const wizard: AzureWizard<IConnectToGitHubWizardContext> = new AzureWizard(context, {
+        title,
+        promptSteps: [
+            new GitHubOrgListStep(),
+            new GitHubRepoListStep(),
+            new GitHubBranchListStep()
+        ]
+    });
+
+    try {
+        await wizard.prompt();
+    } catch (error) {
+        const parsedError: IParsedError = parseError(error);
+        if (parsedError.message.indexOf('Bad credentials') > -1) {
```

Comment:
Why did you move this logic outside getGitHubJsonResponse? If you ever change this to be a subwizard instead of it's own wizard - you'd have to move this logic back since you wouldn't call wizard.prompt() directly.

Reply:
Well, it was because it was the only reason each step required the node and client to be passed in which seemed excessive.

Reply:
Ah well I'm fine with the change in that case (although like I said you might have to refactor if you ever move this to a subwizard)

Reply:
Future Nathan's problem 😼